pbj-core/pbj-grpc-client-helidon/src/main/java/com/hedera/pbj/grpc/client/helidon/PbjGrpcCall.java

```diff
@@ -173,13 +192,23 @@ private void receiveRepliesLoop() {
                     datagramReader.add(bufferData);
 
                     // ...and then feed all complete GRPC datagrams to the pipeline:
-                    BufferData data;
-                    while ((data = datagramReader.extractNextDatagram()) != null) {
+                    PbjGrpcDatagramReader.Datagram datagram;
+                    while ((datagram = datagramReader.extractNextDatagram()) != null) {
+                        if (datagram.compressedFlag() != 0 && datagram.compressedFlag() != 1) {
+                            throw new IllegalStateException("GRPC datagram compressed flag " + datagram.compressedFlag()
+                                    + " is unsupported. Only 0 and 1 are valid.");
+                        }
+                        BufferData data = datagram.data();
                         final byte[] array = data.readBytes();
                         final Bytes bytes = Bytes.wrap(array);
+                        // If the compressedFlag is 0, then per the specification, the message isn't compressed
```

Comment:
It makes sense to check the opposite: the flag is set, but decompressor is null (or IDENTITY). Today it will just throw an NPE at line 208 below, but it can be changed to a more useful exception